wg-api/spec-documents/scroll-view.md

```diff
@@ -0,0 +1,281 @@
+
+Controls how the horizontal scroll bar appears and functions.
+* `disabled` - The scrollbar is hidden.
+* `enabled-but-hidden` - The scrollbar is hidden whether or not the contents are larger than the viewport, but the pane will respond to scroll events. _Linux_  _Windows_
+*`enabled` - The scrollbar will be visible if the contents are larger than the viewport.
+
+#### `scrollView.setVerticalScrollBarMode(mode)` _Experimental_
+
+* `mode` string - Can be `disabled`, `enabled-but-hidden`, `enabled`. Default is `enabled`.
+
+Controls how the vertical scroll bar appears and functions.
+
+#### `scrollView.getHorizontalScrollBarMode()` _Experimental_
+
+Returns `string` - horizontal scrollbar mode.
+
+#### `scrollView.getVerticalScrollBarMode()` _Experimental_
+
+Returns `string` - vertical scrollbar mode.
+
+#### `scrollView.setHorizontalScrollElasticity(elasticity)` _macOS_ _Experimental_
+
+* `elasticity` string - Can be `automatic`, `none`, `allowed`. Default is `automatic`.
+
+The scroll view’s horizontal scrolling elasticity mode.
+A scroll view can scroll its contents past its bounds to achieve an elastic effect. 
+When set to `automatic`, scrolling the horizontal axis beyond its document
+bounds only occurs if the document width is greater than the view width,
+or the vertical scroller is hidden and the horizontal scroller is visible.
+* `automatic` - Automatically determine whether to allow elasticity on this axis.
+* `none` - Disallow scrolling beyond document bounds on this axis.
+* `allowed` - Allow content to be scrolled past its bounds on this axis in an elastic fashion.
+
+#### `scrollView.setVerticalScrollElasticity(elasticity)` _macOS_ _Experimental_
+
+* `elasticity` string - Can be `automatic`, `none`, `allowed`. Default is `automatic`.
+
+The scroll view’s vertical scrolling elasticity mode.
+
+#### `scrollView.getHorizontalScrollElasticity()` _macOS_ _Experimental_
+
+Returns `string` - The scroll view’s horizontal scrolling elasticity mode.
+
+#### `scrollView.getVerticalScrollElasticity()` _macOS_ _Experimental_
+
+Returns `string` - The scroll view’s vertical scrolling elasticity mode.
+
+#### `view.setScrollPosition(point)` _Experimental_
+
+* `point` - The point in the `contentView` to scroll to.
+
+Scrolls the view’s closest ancestor `clipView` object so a point in the
+view lies at the origin of the clip view's bounds rectangle.
+
+#### `view.getScrollPosition()` _Experimental_
+
+Returns `Point`
+
+#### `view.getMaximumScrollPosition()` _Experimental_
+
+Returns `Point`
+
+#### Events
+
+#### Event: 'did-scroll' _Experimental_
+
+Returns:
+
+* `event` Event
+
+Emitted when the content view is being scrolled.
+
+#### Event: 'start-scroll' _Experimental_
+
+Returns:
+
+* `event` Event
+
+Emitted at the beginning of user-initiated scroll tracking.
+
+#### Event: 'end-scroll' _Experimental_
+
+Returns:
+
+* `event` Event
+
+Emitted at the end of scroll tracking.
```

Comment:
* `did-scroll` and `end-scroll` are pretty similar in naming. Maybe something like `will-scroll` (before), `scroll` (during), and `did-scroll` (after) would work better?

* More detail for each event similar to https://www.electronjs.org/docs/latest/api/web-contents#event-will-navigate would be helpful. For instance, will the scroll be able to be cancelled via `event.preventDefault`?

Reply:
This is for revisit in future - as currently taken from Apple documentation, it will need some adjustment when implementing by Chromium views.

Reply:
Do you mean the naming convention is from Apple? Is this specific to Cocoa? Sorry if I'm misunderstanding!

Reply:
Not really; slighty adjusted in 4b573a32b643c8f224c418b9aa83a12c65b7e210